include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -0,0 +1,356 @@
+/*
+  Copyright (C) 2011 - 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__velocity_boundary_conditions_ascii_data_h
+#define __aspect__velocity_boundary_conditions_ascii_data_h
+
+#include <aspect/velocity_boundary_conditions/interface.h>
+#include <aspect/simulator_access.h>
+
+
+namespace aspect
+{
+  namespace VelocityBoundaryConditions
+  {
+    using namespace dealii;
+
+
+    namespace internal
+    {
+
+      /**
+       * AsciiDataLookup handles all kinds of tasks around looking up a certain
+       * velocity boundary condition from a text file. This class
+       * keeps around the contents of two sets of files, corresponding to two
+       * instances in time where the text files provides us with data; the boundary
+       * values at one particular time are interpolated between the two
+       * currently loaded data sets.
+       */
+      template <int dim>
+      class AsciiDataLookup
+      {
+        public:
+
+          /**
+           * Initialize all members and the two tables referring to the
+           * actual velocities.
+           */
+          AsciiDataLookup(const Point<dim> &grid_extent_,
+                          const unsigned int npoints_[3],
+                          const unsigned int boundary_dimensions_[2],
```

Comment:
I tripped myself up. The correct syntax is
```
  const unsigned int (&n_points)[3]
```
i.e., a reference to an array of 3 elements.